### PR TITLE
Add service monitors

### PR DESCRIPTION
**Pull request by ndragon798**

### CODE REVIEW

File: charts/immich/Chart.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 apiVersion: v2
 description: A chart to power Immich (immich.app) running on kubernetes
 name: immich
-version: 0.4.0
-appVersion: v1.95.1
+version: 0.4.1
```
Comment (bo0tzz) [inline charts/immich/Chart.yaml:5]: I think this warrants a minor version bump.

```suggestion
version: 0.5.0
```

File: charts/immich/templates/microservices.yaml [hunk 2]
```diff
@@ -8,7 +8,22 @@ args:
 
 service:
   main:
```
Comment (bo0tzz) [inline charts/immich/templates/microservices.yaml:10]: nit: let's call the service `metrics`

File: charts/immich/templates/microservices.yaml [hunk 3]
```diff
@@ -8,7 +8,22 @@ args:
...
+    ports:
+      http: 
+        enabled: true
+        primary: true
+        port: 8081
+        protocol: HTTP
+
+serviceMonitor:
+  main:
+    enabled: true
```
Comment (bo0tzz) [inline charts/immich/templates/microservices.yaml:23]: Should we leave this disabled by default?

**Comment by ndragon798** [comment]
https://github.com/immich-app/immich-charts/issues/73 should close this issue

File: charts/immich/templates/microservices.yaml [hunk 4]
```diff
@@ -8,7 +8,22 @@ args:
 
 service:
   main:
-    enabled: false
+    enabled: true
+    primary: true
```
Comment (bo0tzz) [inline charts/immich/templates/microservices.yaml:12]: I don't think this svc should be `primary`

File: charts/immich/templates/server.yaml [hunk 5]
```diff
@@ -17,14 +17,25 @@ service:
         primary: true
         port: 3001
         protocol: HTTP
+      mon:
```
Comment (bo0tzz) [inline charts/immich/templates/server.yaml:20]:
```suggestion
      metrics:
```

**Review by bo0tzz — CHANGES_REQUESTED** [review]
I'd been meaning to add this, thank you! I have a few comments but they're pretty much all nitpicks, so don't be too alarmed :)

Can you also update the values.yaml to
1. Add the `IMMICH_METRICS` env var (probably false by default?)
2. Bump the image tag to v1.100.0

File: charts/immich/templates/microservices.yaml (hunk 2 above)
Comment (ndragon798) [reply charts/immich/templates/microservices.yaml:10]: Can it be changed from main? I think main has to be there or else I get an error about the ports not being defined.
```
Release "immich-test" does not exist. Installing it now.
Error: 2 errors occurred:
	* Service "immich-test-microservices" is invalid: spec.ports[0].port: Invalid value: 0: must be between 1 and 65535, inclusive
	* Deployment.apps "immich-test-microservices" is invalid: [spec.template.spec.containers[0].ports[0].containerPort: Required value, spec.template.spec.containers[0].ports[1].name: Duplicate value: "http", spec.template.spec.containers[0].livenessProbe.tcpSocket.port: Invalid value: 0: must be between 1 and 65535, inclusive, spec.template.spec.containers[0].readinessProbe.tcpSocket.port: Invalid value: 0: must be between 1 and 65535, inclusive, spec.template.spec.containers[0].startupProbe.tcpSocket.port: Invalid value: 0: must be between 1 and 65535, inclusive]

```

Comment (ndragon798) [reply charts/immich/templates/microservices.yaml:15]: Resolved on discord

**Comment by ndragon798** [comment]
Don't merge this yet I just saw that for some reason im overwriting the env vars on the server when the metrics are enabled.

**Review by bo0tzz — APPROVED** [review]

**Pull request merged**